Extract HttpUserAgent class from HttpSM

JosiahWI opened the pull request:
This extracts a new class with 3 fields (entry, raw_buffer_reader, and txn). It doesn't have much responsibility yet. But it does provide a starting point for moving some behavior out of the state machine.

JosiahWI committed:
Extract HttpUserAgent class from HttpSM

This extracts a new class with 3 fields (entry, raw_buffer_reader,
and txn). It doesn't have much responsibility yet. But it does
provide a starting point for moving some behavior out of the
state machine.

bneradt commented on proxy/http/HttpUserAgent.cc:
```diff
@@ -0,0 +1,63 @@
+#include "I_IOBuffer.h"
+#include "ProxyTransaction.h"
+
+HttpVCTableEntry *
+HttpUserAgent::get_entry() const
+{
+  return m_entry;
+}
+
+void
+HttpUserAgent::set_entry(HttpVCTableEntry *entry)
+{
+  m_entry = entry;
+}
+
+IOBufferReader *
+HttpUserAgent::get_raw_buffer_reader()
+{
+  return m_raw_buffer_reader;
+}
+
+void
+HttpUserAgent::set_raw_buffer_reader(IOBufferReader *raw_buffer_reader)
+{
+  m_raw_buffer_reader = raw_buffer_reader;
+}
+
+ProxyTransaction *
+HttpUserAgent::get_txn() const
+{
+  return m_txn;
+}
+
+void
+HttpUserAgent::set_txn(ProxyTransaction *txn)
+{
+  m_txn = txn;
+}
```
This is pretty hot code. We probably need to keep these inlined in the header file. (Declare inline, and provide definitions at the bottom of the header file.)

bneradt reviewed (changes requested):
Looks good.

JosiahWI committed:
Move virtual connection stuff into HttpVCTable.h

This introduces HttpVCTable.h and HttpVCTable.cc to hold all the
stuff related to the HttpVCTable struct.

JosiahWI committed:
Move a bunch of client_ fields into HttpUserAgent

This moves all the client_ fields except the byte counts into
HttpUserAgent, and sets most of the fields when the transaction
is set.

JosiahWI committed:
Reset some client fields to defaults in set_txn

This matches the behavior of the original code, and may be
important.

JosiahWI committed:
WIP Fix incorrect tcp_reused detection

A not (!) operator was missed, which caused tcp_reused to be
set incorrectly on the user agent. This fixes the issue, and adds
Catch tests to check for regression.

JosiahWI committed:
Clean up debugging changes

There is one segfault left to fix.

JosiahWI committed:
Fix format

JosiahWI committed:
Fix proxy/http Makefile

JosiahWI committed:
Add test_error_page_selection to CMake

JosiahWI committed:
Fix CMake build to install traffic_ctl target (#9996)

The CMakeLists in src/traffic_ctl was erroneously installing the
traffic_server target instead of the traffic_ctl target.

JosiahWI committed:
Fix rebase errors

JosiahWI committed:
Add a getter for the state machine's user agent

JosiahWI committed:
Test against UserAgent instead of HttpSM

JosiahWI committed:
Move test_HttpSM to test_HttpUserAgent

JosiahWI committed:
Do not build test_HttpUserAgent in automake build

JosiahWI committed:
Add new sources back to automake
